presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcPageSource.java

```diff
@@ -233,5 +243,31 @@ public String toString()
                     .add("index", index)
                     .toString();
         }
+
+        private static final class MaskingBlockLoader
+                implements LazyBlockLoader
+        {
+            private Function<Block, Block> maskDeletedRowsFunction;
```

Comment:
I think there's isn't much advantage from `MaskDeletedRowsFunction` implementing `Function` interface.
I'd use `MaskDeletedRowsFunction` here & unimplement the interface, to make code navigation easier.

Reply:
I agree with @findepi on the part of easier code navigation but will defer to @dain on this one too

Reply:
I don't care either way